static/js/timerender.js

```diff
@@ -179,13 +179,26 @@ export function render_date(time, time_above, today) {
 }
 
 // Renders the timestamp returned by the <time:> Markdown syntax.
-export function render_markdown_timestamp(time, text) {
+export function render_markdown_timestamp(time) {
     const hourformat = page_params.twenty_four_hour_time ? "HH:mm" : "h:mm a";
     const timestring = format(time, "E, MMM d yyyy, " + hourformat);
-    const titlestring = "This time is in your timezone. Original text was '" + text + "'.";
+    const timezone = new Intl.DateTimeFormat(undefined, {timeZoneName: "short"})
+        .formatToParts(time)
+        .find(({type}) => type === "timeZoneName").value;
+
+    const tz_offset = -time.getTimezoneOffset() / 60;
+    const tz_offset_str =
+        timezone === "UTC" ? "" : " (UTC" + (tz_offset >= 0 ? "+" : "-") + tz_offset + ")";
```

Comment:
I also noticed a bug that PST gets displayed as `Your time zone: PDT (UTC--7)` for me -- that double negative seems like a bug.

I think we should extract the logic for producing the timezone to display to a function that just does that, rather than duplicating it between here are the `message_time` tooltip, and then write a unit test suite for it with PST included in it, so that we're setup to readily fix any future bugs of this form that we find.  

Then this function can mostly just call that and then rendering a small template for generating the 2-line HTML structure.

Maybe we can call that template `static/templates/tooltips/markdown_time_element.hbs`

Reply:
From my [comment](https://github.com/zulip/zulip/pull/19026#pullrequestreview-697689895) on the other PR:

> Also, the standard format for fractional-hour zones is “UTC+05:30”, not “UTC+5.5” (and especially not “[UTC+5.666666666666667](https://en.wikipedia.org/wiki/UTC%2B05:40)”!).

Reply:
Extracted that logic to a function. See 2nd commit. Now, it shows “UTC+05:30” instead of “UTC+5.5” and also not “UTC+5.666666666666667".  @andersk Can you take a look?